cmk/special_agents/agent_fuse_management_central.py

```diff
@@ -0,0 +1,265 @@
+            )
+            if (system["id"], component["id"]) in system_alerts_map:
+                alert = system_alerts_map[(system["id"], component["id"])]
+                if "errors" in alert:
+                    system_alert.errors = int(alert["errors"])
+                if "warnings" in alert:
+                    system_alert.warnings = int(alert["warnings"])
+                if "link" in alert:
+                    system_alert.link = alert["link"]
+            system_alerts.append(system_alert)
+    return system_alerts
+
+
+def get_environment_alerts(layout: dict, env_alert_map: Dict[Tuple[str, str], dict]) -> List[Alert]:
+    env_alerts: List[Alert] = []
+    if "environments" not in layout:
+        return env_alerts
+
+    for environment in layout["environments"]:
+        for component in environment["componentTypes"]:
+            env_alert: Alert = Alert(
+                environment["id"], environment["name"], "", component["displayName"], 0, 0, ""
+            )
+            if (environment["id"], component["id"]) in env_alert_map:
+                alert = env_alert_map[(environment["id"], component["id"])]
+                if "errors" in alert:
+                    env_alert.errors = int(alert["errors"])
+                if "warnings" in alert:
+                    env_alert.warnings = int(alert["warnings"])
+                if "link" in alert:
+                    env_alert.link = alert["link"]
+            env_alerts.append(env_alert)
+    return env_alerts
+
+
+def get_admin_alerts(layout: dict, env_admin_map: Dict[str, dict]) -> List[Alert]:
+    admin_alerts: List[Alert] = []
+    if "admin" not in layout:
+        return admin_alerts
+
+    admin = layout["admin"]
+    for component in admin["componentTypes"]:
+        admin_alert: Alert = Alert("", "", "", component["displayName"], 0, 0, "")
+        if component["id"] in env_admin_map:
+            alert = env_admin_map[component["id"]]
+            if "errors" in alert:
+                admin_alert.errors = int(alert["errors"])
+            if "warnings" in alert:
+                admin_alert.warnings = int(alert["warnings"])
+            if "link" in alert:
+                admin_alert.link = alert["link"]
+        admin_alerts.append(admin_alert)
+    return admin_alerts
+
+
+def parse_arguments(argv: List[str]) -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("user", metavar="USER", help="")
+    parser.add_argument("password", metavar="PASSWORD", help="")
+    parser.add_argument("url", metavar="URL", help="")
+    parser.add_argument("host", metavar="HOST", help="")
+    parser.add_argument(
+        "--debug", action="store_true", help="Debug mode: let Python exceptions raise through"
+    )
+    return parser.parse_args(argv)
+
+
+def get_state(layout_response_code: int, summary_response_code: int, data) -> str:
```

Comment:
I think you want to make this a service later. In that case, it is better to give out the raw information here and let the check decide what to do with it later. It means more flexibility and also that the evaluation is always in the same place.